implementation/src/main/java/io/smallrye/mutiny/operators/uni/builders/UniJoinAll.java

```diff
@@ -45,22 +47,33 @@ public void subscribe(UniSubscriber<? super List<T>> subscriber) {
 
         private final AtomicReferenceArray<T> items = new AtomicReferenceArray<>(unis.size());
         private final List<Throwable> failures = Collections.synchronizedList(new ArrayList<>());
-        private final AtomicInteger signalsCount = new AtomicInteger();
+        private final AtomicInteger completionSignalsCount = new AtomicInteger();
+
+        private AtomicInteger nextSubscriptionIndex;
 
         public UniJoinAllSubscription(UniSubscriber<? super List<T>> subscriber) {
             this.subscriber = subscriber;
         }
 
         public void triggerSubscriptions() {
-            for (int i = 0; i < unis.size() && !cancelled.get(); i++) {
-                int index = i;
-                Uni<? extends T> uni = unis.get(i);
-                uni.onSubscription()
-                        .invoke(subscription -> subscriptions.set(index, subscription))
-                        .subscribe().with(subscriber.context(), item -> this.onItem(index, item), this::onFailure);
+            int limit;
+            if (concurrency != -1) {
+                limit = Math.min(concurrency, unis.size());
+                nextSubscriptionIndex = new AtomicInteger(concurrency - 1);
+            } else {
+                limit = unis.size();
+            }
```

Comment:
Cosmetic: Extra line

Reply:
If you increase your browser width it should not appear as an extra line 😄